src/modules/ModuleGovernance/utils.ts

```diff
@@ -0,0 +1,31 @@
+import { POSITIVE_CHOICES } from './const'
+import { ProposalState, ProposalStatus, RawProposal } from './types'
+
+export function getLowerCaseChoice(choice: string) {
+  return choice.toLocaleLowerCase().trim()
+}
+
+export function getLowerCaseChoices(choices: RawProposal['choices']) {
+  return choices.map(choice => getLowerCaseChoice(choice))
+}
+
+export function getProposalStatus({ state, choices, scores }: Pick<RawProposal, 'state' | 'choices' | 'scores'>): ProposalStatus {
+  let status: ProposalStatus | null = null
+  if (state === ProposalState.Active)
+    status = ProposalStatus.Active
+  else {
+    const positiveChoices = getLowerCaseChoices(POSITIVE_CHOICES)
+    const proposalChoicesInLowerCase = getLowerCaseChoices(choices)
+    const hasPositiveChoice = positiveChoices.some(positiveChoice => proposalChoicesInLowerCase.includes(positiveChoice))
+    if (hasPositiveChoice) {
+      const maxScore = Math.max(...scores)
+      const resultChoice = proposalChoicesInLowerCase[scores.indexOf(maxScore)]
+      if (positiveChoices.includes(resultChoice))
+        status = ProposalStatus.Executed
+      else
+        status = ProposalStatus.Defeated
+    } else
+      status = ProposalStatus.Finished
+  }
+  return status
+}
```

Comment:
Would be nice to have basic unit tests around it.

Reply:
Ok, but we should discuss this solution, in general, in the beginning. I am not sure that it is correct. I'll make a unit test later